### PR TITLE
PLG-642: Add key figure component in DSM

### PR DESCRIPTION
**Pull request by rouliane**
image

### CODE REVIEW

File: front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  const valuesWithLabels: FigureWithLabelProps[] = [];
+  let simpleFigure: ReactElement | null = null;
+
+  React.Children.forEach(children, child => {
+    if (isValidElement<FigureWithLabelProps>(child) && FigureWithLabel === child.type) {
+      valuesWithLabels.push(child);
+    }
+  });
+  React.Children.forEach(children, child => {
+    if (isValidElement(child) && Figure === child.type) {
```
Comment (BryanS29-akeneo) [inline front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.tsx:97]: can't we put this condition in the foreach above ?

File: front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    {args => {
+      return (
+        <KeyFigureGrid>
+          <KeyFigure icon={React.createElement(Icons[args.icon])} title="Key figure 1">
+            <KeyFigure.FigureWithLabel>
+              Average: <KeyFigure.Figure>10</KeyFigure.Figure>
+            </KeyFigure.FigureWithLabel>
+            <KeyFigure.FigureWithLabel>
+              Max: <KeyFigure.Figure>15</KeyFigure.Figure>
+            </KeyFigure.FigureWithLabel>
```
Comment (olivierpontier) [inline front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.stories.mdx:80]:
```suggestion
              Average: <KeyFigure.Figure>10</KeyFigure.Figure>
              Max: <KeyFigure.Figure>15</KeyFigure.Figure>
```

**Review by olivierpontier — APPROVED** [review]
GTM after adding unit tests

**Review by LaurentPetard — APPROVED** [review]

**Review by BryanS29-akeneo — APPROVED** [review]

File: front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+## Playground
+
+<Canvas>
+  <Story name="Standard">
+    {args => {
+      return (
+        <KeyFigureGrid>
+          <KeyFigure icon={React.createElement(Icons[args.icon])} title={args.title}>
+            Average: <KeyFigure.Figure>10</KeyFigure.Figure>
```
Comment (juliensnz) [inline front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.stories.mdx:32]: What about 
```suggestion
            <KeyFigure.Figure label="Average">10</KeyFigure.Figure>
```

File: front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            Max: <KeyFigure.Figure>15</KeyFigure.Figure>
```
Comment (juliensnz) [inline front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.stories.mdx:33]:
```suggestion
            <KeyFigure.Figure label="Max">15</KeyFigure.Figure>
```

File: front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  text-overflow: ellipsis;
+`;
+const Values = styled.div`
+  display: flex;
+  color: ${getColor('grey', 100)};
+  font-size: 16px;
+`;
+
+const KeyFigure = ({icon, title, children}: KeyFigureProps) => {
+  const validIcon = isValidElement<IconProps>(icon) && React.cloneElement(icon, {size: 30});
```
Comment (juliensnz) [inline front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.tsx:64]: maybe you could pass the svg color here instead of css?

**Review by juliensnz — APPROVED** [review]
I did small suggestion :)

File: front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
+import React, {isValidElement, ReactElement, ReactNode} from 'react';
+import styled from 'styled-components';
+import {getColor, getFontSize} from '../../theme';
+import {IconProps} from '../../icons';
+
+const Figure = styled.div`
+  color: ${getColor('brand', 100)};
+  font-size: 16px;
```
Comment (pierallard) [inline front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.tsx:8]: Please use the getFontSize

File: front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.tsx [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+  color: ${getColor('grey', 140)};
+  font-size: ${getFontSize('big')};
+  overflow: hidden;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+`;
+const Values = styled.div`
+  display: flex;
+  color: ${getColor('grey', 100)};
+  font-size: 16px;
```
Comment (pierallard) [inline front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.tsx:60]: same

File: front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.tsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+const KeyFigureGrid = styled.div`
+  display: grid;
+  grid-template-columns: repeat(auto-fit, minmax(320px, 1fr));
+  gap: 20px;
+`;
+
+KeyFigure.Figure = Figure;
+
+export {KeyFigure, KeyFigureGrid};
```
Comment (pierallard) [inline front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.tsx:85]: Why not export KeyFigureGrid as KeyFigure.Grid ? (Suggestion, not sure)

File: front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.stories.mdx (hunk 3 above)
Comment (rouliane) [reply front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.stories.mdx:32]: It's a better idea! I'll fix it this way.

File: front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.stories.mdx [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+<Canvas>
+  <Story name="KeyFigure grid">
+    {args => {
+      return (
+        <KeyFigureGrid>
+          <KeyFigure icon={React.createElement(Icons[args.icon])} title="Key figure 1">
+            Average: <KeyFigure.Figure>10</KeyFigure.Figure>
+            Max: <KeyFigure.Figure>15</KeyFigure.Figure>
+          </KeyFigure>
+          <KeyFigure icon={React.createElement(Icons[args.icon])} title="Key figure 2 with a very very very very very long label">
```
Comment (pierallard) [inline front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.stories.mdx:70]: Can you check it works with a KeyFigure.Figure very long as above ?

**Review by pierallard — APPROVED** [review]

File: front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.tsx (hunk 6 above)
Comment (rouliane) [reply front-packages/akeneo-design-system/src/components/KeyFigure/KeyFigure.tsx:8]: There is no pre-defined font size for 16px, I wouldn't know how to call it

**Pull request merged**